components/job-orchestration/job_orchestration/executor/search/fs_search_task.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import os
+import sys
+import signal
+import subprocess
+from pathlib import Path
+from typing import Any, Dict
+
+from celery.app.task import Task
+from celery.utils.log import get_task_logger
+
+from clp_py_utils.clp_logging import get_logging_formatter, set_logging_level
+
+from job_orchestration.executor.search.celery import app
+from job_orchestration.job_config import SearchConfig
+
+
+logger = get_task_logger(__name__)
+
+@app.task(bind=True)
+def search(
+    self: Task,
+    job_id: str,
+    search_config_obj: dict,
+    archive_id: str,
+    results_cache_uri: str,
+) -> Dict[str, Any]:
+    task_id = str(self.request.id)
+    clp_home = Path(os.getenv("CLP_HOME"))
+    archive_directory = Path(os.getenv('CLP_ARCHIVE_OUTPUT_DIR'))
+    clp_logs_dir = Path(os.getenv("CLP_LOGS_DIR"))
+    clp_logging_level = str(os.getenv("CLP_LOGGING_LEVEL"))
+
+    # Setup logging to file
+    worker_logs_dir = clp_logs_dir / job_id
+    worker_logs_dir.mkdir(exist_ok=True, parents=True)
+    worker_logs = worker_logs_dir / f"{task_id}.log"
+    logging_file_handler = logging.FileHandler(filename=worker_logs, encoding="utf-8")
+    logging_file_handler.setFormatter(get_logging_formatter())
+    logger.addHandler(logging_file_handler)
+    set_logging_level(logger, clp_logging_level)
+    stderr_log_path = worker_logs_dir / f"{task_id}-stderr.log"
+    stderr_log_file = open(stderr_log_path, "w")
+
+    logger.info(f"Started job {job_id}. Task Id={task_id}.")
```

Comment:
Not sure how easy/doable that is with the logger celery is handing us on line 18? Might make sense to use our own logger instead of celery's logger though -- can take a look.

Reply:
So it seems like get_task_logger yields a logger that sends logs to the same file for each worker. The format in that file does contain the "task id".

Might be best to simplify by getting rid of all of this per-worker setup where we log to a different directory per job and leaving celery's task logger to work how it is supposed to.

Reply:
Granted that doesn't work for logs coming from clo, so maybe we should still create a per-task logfile for clo to write to?

Reply:
Could possibly also just open one file per job in append mode that each search task in a job makes clo write to?

Reply:
Gotcha. Yeah, perhaps we can just log clo's output to a file per task in the same dir as Celery is writing logs to?

Reply:
> Could possibly also just open one file per job in append mode that each search task in a job makes clo write to?

Er, how would that work? Do we need to do it through Celery?

Reply:
For now I'm letting celery's task logger do its default behaviour, and creating one file per task for clo to write to. We may want to change this behaviour later to configure clo to emit logs containing the task id, and have each clo process append to the same file.

Reply:
> > Could possibly also just open one file per job in append mode that each search task in a job makes clo write to?
> 
> Er, how would that work? Do we need to do it through Celery?

Doing it by creating it ourselves within the search task, and having search tasks for each job open the same file in append mode, and passing that fd to clo.

Reply:
> For now I'm letting celery's task logger do its default behaviour, and creating one file per task for clo to write to. We may want to change this behaviour later to configure clo to emit logs containing the task id, and have each clo process append to the same file.

To be clear what I'm doing now is creating a subdirectory in the workers log directory per job, and having each task open it's own log file per task in write mode in that directory which gets passed to clo.